livedata/src/main/java/io/getstream/chat/android/livedata/controller/QueryChannelsControllerImpl.kt

```diff
@@ -199,9 +199,9 @@ class QueryChannelsControllerImpl(
     private fun addChannels(newChannels: List<Channel>, onTop: Boolean = false) {
         // second page adds to the list of channels
         queryEntity.channelCids = if (onTop) {
-            newChannels.map { it.cid } + queryEntity.channelCids
+            (newChannels.map { it.cid } + queryEntity.channelCids).distinct()
         } else {
-            queryEntity.channelCids + newChannels.map { it.cid }
+            (queryEntity.channelCids + newChannels.map { it.cid }).distinct()
```

Comment:
In the first branch, any newly added channels that were already present in the list will move to the top (because `distinct` will remove their second occurrence, which is their original position). That's intentional, right?

In the second branch, we're simply adding fewer channels to the end of the list this way, and the existing ones don't get modified / moved around.

Reply:
I can't answer this question properly from my current level of knowledge how the system should works. 
Doubles could happen when you had created new channel locally and then received event from backend that channel was created (really the same one). Usually it is already on top of the channel list. 
And if the order of such duplication really matters there is must be test for that.

Reply:
@JcMinarro what do you think?

Reply:
I think it is fine, When some new channel is added to the top of the list it is because new event is received (new channel created, channel receive a new message and so on). We are getting the id to perform a query over it (Some sync on background) and we want to keeps this order